sympy/solvers/ode.py

```diff
@@ -4252,19 +4266,30 @@ def _is_special_case_of(soln1, soln2, eq, order, var):
     # means that some value of the constants in sol1 is a special case of
     # sol2 corresponding to a particular choice of the integration constants.
 
+    if isinstance(soln1, Eq):
+        soln1 = soln1.rhs - soln1.lhs
+    if isinstance(soln2, Eq):
+        soln2 = soln2.rhs - soln2.lhs
```

Comment:
Is the solution sometimes not an Eq?

Reply:
yes

Reply:
Given the changes you've made below is the solution not always an Eq now?